src/features/view-nft/Actions/Actions.tsx

```diff
@@ -1,91 +1,71 @@
-import { Labels } from '../../../lib/constants/labels';
-
 import { useWeb3 } from '../../../lib/hooks/useWeb3';
 import { useBuyNowPrice } from '../../../lib/hooks/useBuyNowPrice';
 import { useBidData } from '../../../lib/hooks/useBidData';
+import { useDomainMetrics } from '../../../lib/hooks/useDomainMetrics';
+import { formatEthers, formatNumber } from '../../../lib/util/number/number';
+import { getUserBids } from '../../../lib/util/bids/bids';
+import { Metadata } from '../../../lib/types/metadata';
+import { Labels } from '../../../lib/constants/labels';
+import { Domain } from '@zero-tech/zns-sdk';
 
-import {
-	getOrderedActions,
-	getUsdConversion,
-	getVisibleActions,
-} from './Actions.utils';
-import { formatNumber } from '../../../lib/util/number/number';
-import { getHighestBid, getUserBids } from '../../../lib/util/bids/bids';
-
+import { DataTestId } from './Actions.constants';
 import { ActionBlock, ActionTypes } from './Actions.types';
-import { Metadata } from '../../../lib/types/metadata';
+import { getOrderedActions, getVisibleActions } from './Actions.utils';
 
 import styles from './Actions.module.scss';
 
 type ActionsProps = {
 	domain?: Domain;
 	domainMetadata?: Metadata;
-	paymentTokenInfo?: TokenPriceInfo;
 };
 
-export const Actions = ({
-	domain,
-	domainMetadata,
-	paymentTokenInfo,
-}: ActionsProps) => {
+export const Actions = ({ domain, domainMetadata }: ActionsProps) => {
 	const { account } = useWeb3();
-	const { data: buyNowPrice } = useBuyNowPrice(domain?.id);
+	const { data: buyNowPriceData } = useBuyNowPrice(domain?.id);
+	const { data: metrics } = useDomainMetrics(domain?.id);
 	const { data: bids } = useBidData(domain?.id);
-	const userBids = getUserBids(account, bids);
-	const highestBid = getHighestBid(bids);
-	const highestUserBid = getHighestBid(userBids);
+
+	const buyNowPrice = buyNowPriceData ? formatNumber(buyNowPriceData) : '-';
+	const userBids = getUserBids(account, bids) ?? [];
+	const highestUserBid =
+		userBids.length > 0 ? formatEthers(userBids[0]?.amount) : '-';
+	const highestBid =
+		metrics?.highestBid > '0' ? formatEthers(metrics?.highestBid) : '-';
+
 	const isOwnedByUser = domain?.owner?.toLowerCase() === account?.toLowerCase();
 	const isBiddable = !isOwnedByUser || Boolean(domainMetadata?.isBiddable);
-	const isUserBid = !isOwnedByUser && Boolean(highestUserBid);
+	const isUserBid = !isOwnedByUser && userBids.length > 0;
 	const isSetBuyNow = isOwnedByUser && Boolean(domain?.name);
 	const isBuyNow =
-		Boolean(buyNowPrice) && !isOwnedByUser && Boolean(domain?.name);
+		Boolean(buyNowPriceData) && !isOwnedByUser && Boolean(domain?.name);
 	const isViewBids =
 		isOwnedByUser !== undefined && isBiddable && bids?.length > 0;
 
 	const actions: { [action in ActionTypes]: ActionBlock } = {
 		[ActionTypes.BUY_NOW]: {
-			label: `${Labels.BUY_NOW} (${paymentTokenInfo?.name})`,
-			amountToken: formatNumber(buyNowPrice),
-			amountUsd: getUsdConversion(
```

Comment:
removed these `usd` conversions for now as payment token symbol is going to be undefined